mesh/mesh-config-json.c

```diff
@@ -86,13 +86,47 @@ static bool save_config(json_object *jnode, const char *fname)
 static bool get_int(json_object *jobj, const char *keyword, int *value)
 {
 	json_object *jvalue;
+	json_type jtype;
 
 	if (!json_object_object_get_ex(jobj, keyword, &jvalue))
 		return false;
 
-	*value = json_object_get_int(jvalue);
-	if (errno == EINVAL)
-		return false;
+	jtype = json_object_get_type(jvalue);
+
+	/* To keep backward compatibility with previous configuration format */
+	switch (jtype)
+	{
+		case json_type_int:
+			*value = json_object_get_int(jvalue);
+			if (errno == EINVAL)
+				return false;
+
+			break;
+
+		case json_type_string: {
+			uint16_t hexstr_value;
+			const char* str;
+
+			str = json_object_get_string(jvalue);
```

Comment:
Can't you just try to read string, and when that fails, read an int, and when that fails return error?

I don't think you need to switch by type

Reply:
I've made a quick test and it doesn't work.

Reply:
```    {
      "index":10,
      "key":"964b460ba882491c9407b4fbf2fee952",
      "keyRefresh":0
    },``` => index == 16

Reply:
Ok, docs say it returns the string representation even if value isn't one. So we need to check the type. Still, I'd drop the switch and use a plain `if`.

Reply:
Check out this version, it should be ok.